test/Microsoft.Azure.WebJobs.Host.EndToEndTests/DispatchQueueEndToEndTests.cs

```diff
@@ -0,0 +1,149 @@
+﻿using System;
+using System.Collections.Concurrent;
+using System.Collections.Generic;
+using System.Threading.Tasks;
+using Microsoft.Azure.WebJobs.Host.TestCommon;
+using Microsoft.WindowsAzure.Storage;
+using Microsoft.WindowsAzure.Storage.Queue;
+using Newtonsoft.Json.Linq;
+using Newtonsoft.Json;
+using Xunit;
+
+namespace Microsoft.Azure.WebJobs.Host.EndToEndTests
+{
+    public class DispatchQueueEndToEndTests : IDisposable
+    {
+        private static ConcurrentDictionary<string, int> _concurrent = new ConcurrentDictionary<string, int>();
+        private JobHostConfiguration _hostConfiguration;
+        private const string HOSTID = "testhost";
+        private CloudQueue _sharedQueue;
+        private CloudQueue _poisonQueue;
+
+        public DispatchQueueEndToEndTests()
+        {
+            _hostConfiguration = new JobHostConfiguration();
+            _hostConfiguration.AddExtension(new DispatchQueueTestConfig());
+            _hostConfiguration.HostId = HOSTID;
+            CloudStorageAccount sdkAccount = CloudStorageAccount.Parse(_hostConfiguration.StorageConnectionString);
+            CloudQueueClient client = sdkAccount.CreateCloudQueueClient();
+            _sharedQueue = client.GetQueueReference("azure-webjobs-shared-" + HOSTID);
+            _poisonQueue = client.GetQueueReference("azure-webjobs-poison-" + HOSTID);
```

Comment:
Can we leverage the faked out storage account stuff to avoid a real storage dependency?

Reply:
but this is an end to end test, do we not want to test with actual storage queue at all?
the "phantom" message should be unlikely since I have a Dispose():
 ```
       public void Dispose()
        {
            _sharedQueue.DeleteIfExists();
            _poisonQueue.DeleteIfExists();
        }
```